cpp/src/arrow/util/tdigest.h

```diff
@@ -60,12 +61,27 @@ class ARROW_EXPORT TDigest {
     input_.push_back(value);
   }
 
+  // skip NAN on adding
+  // TODO(yibo): store NAN as is, partition to buffer end before merging
```

Comment:
Why would it be better to partition at the end?

Reply:
No solid benchmark.
The idea is to drop the `isnan` check in Add function. When input buffer is full, partition NaNs to the buffer end and ignore them in sorting and merging. Guess the performance depends heavily on inputs (maybe worse for common cases where NaN is rare and `isnan` branch is always predicted correctly).

Reply:
Removed this comment. Will do benchmark first.